### PR TITLE
Support partition removal

### PR DESCRIPTION
**Pull request by jt2594838**
Partitions can be removed with the following statement:
`DELETE PARTITION root.sg1 0,1,2,3,4` which deletes the first five partitions of storage group root.sg1.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/qp/constant/SQLConstant.java [hunk 1]
```diff
@@ -151,6 +151,7 @@ private SQLConstant() {
   public static final int TOK_LOAD_CONFIGURATION_LOCAL = 86;
 
   public static final int TOK_SHOW_MERGE_STATUS = 87;
+  public static final int TOK_DELETE_PARTITION = 87;
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/qp/constant/SQLConstant.java:154]: still 87?

Comment (jt2594838) [reply server/src/main/java/org/apache/iotdb/db/qp/constant/SQLConstant.java:154]: fixed

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java [hunk 2]
```diff
@@ -2410,4 +2410,58 @@ public boolean isFileAlreadyExist(TsFileResource tsFileResource, long partitionN
...
+   * remove all partitions that satisfy a filter.
+   * @param filter
+   */
+  public void removePartitions(TimePartitionFilter filter) {
+    // this requires blocking all other activities
+    insertLock.writeLock().lock();
+    mergeLock.writeLock().lock();
+    try {
+      // abort ongoing merges
+      MergeManager.getINSTANCE().abortMerge(storageGroupName);
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java:2424]: I notice you add `storageGroupSubTasks` and `storageGroupMainTasks` in 6.9. When to clear the finished tasks?

Comment (jt2594838) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java:2424]: There is a TaskCleanerThread in MergeManager.

**Pull request merged**